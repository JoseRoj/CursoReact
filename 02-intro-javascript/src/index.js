const persona = {
    nombre: 'Jose',
    apellido: 'Rojas',
    edad: 45,
    direccion: {
        ciudad: 'New York',
        zip: 123456,
        lat: 14.3232,
        lng: 34.9233321
    }
}
//!Error: const persona2 = persona;

const persona2 = { ...persona };
persona2.nombre = 'Peter';
console.log(persona);
console.log(persona2);