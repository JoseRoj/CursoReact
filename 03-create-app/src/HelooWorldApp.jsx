/*const newMessage = {
    message: 'Hola Mundo',
    subtitle: 'Subtitulo'
}
const getResult = () => {
    return 'Hola Mundo'
}*/
import PropTypes from 'prop-types'
export const App = ({title, subTitle, name}) => {
    return  (
        <>
            <h1 data-testid = "test-title">{ title}</h1>
            <p> {subTitle}</p>
            <p> {name}</p>
        </>
  )
}

App.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string.isRequired,
    name: PropTypes.string,
}

App.defaultProps = {
    title: 'Hola Mundo',
    subTitle: "No hay subtitulo",
    name: 'Trunks',
}