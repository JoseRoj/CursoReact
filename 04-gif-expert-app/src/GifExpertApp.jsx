import { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState([" "]);

  const onAddCategory = (newCategory) => {
    console.log(newCategory);
    if (categories.includes(newCategory)) return;
    setCategories([newCategory, ...categories]);
    // setCategories(cats => [...cats, "HunterXHunter"]);
  };
  return (
    <>
      <h1>GifExpertApp</h1>
      <AddCategory onNewCategory={onAddCategory} />
      {/*<button onClick={onAddCategory}>Agregar</button>*/}
      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
      </ol>
    </>
  );
};
