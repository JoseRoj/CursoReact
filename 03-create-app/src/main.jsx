import React from 'react'
import ReactDOM from 'react-dom/client'
import { App } from './HelooWorldApp'
import './style.css'
import { CounterApp } from './CounterApp'


ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <CounterApp value={20}> </CounterApp>
    </React.StrictMode>
)