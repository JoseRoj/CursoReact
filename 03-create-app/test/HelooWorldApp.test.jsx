import { App } from "../src/HelooWorldApp";
import {getByTestId, render} from "@testing-library/react";
describe("Pruebas en el componente <HelooWorldApp />", () => {
    test("debe de hacer match con el snapshot", () => {
        const title = "Hola Mundo";
        const { container } = render(<App title= {title}/>);
        expect(container).toMatchSnapshot();
    });
    test("debe de mostrar el titulo en un h1", () => {
        const title = "Hola Mundo";
        const { container, getByText, getByTestId } = render(<App title= {title}/>);
        expect(getByText(title)).toBeTruthy();
        expect(getByTestId("test-title").innerHTML).toContain(title);
        //const h1 = container.querySelector("h1");
        //expect(h1.textContent).toBe(title);
        // expect(h1.textContent).toContain(title);
    });
    test("debe de mostrar el subtitulo envaidado por props", () => {
        const title = 'Hola Mundo';
        const subTitle = 'Subtitulo';
        
        const { getByText, getAllByAltText } = render(<App title={title} subTitle={subTitle} />);
        //expect(getAllByAltText(subTitle).length).toBe(1);
        expect(getByText(subTitle)).toBeTruthy();
    }); 
});