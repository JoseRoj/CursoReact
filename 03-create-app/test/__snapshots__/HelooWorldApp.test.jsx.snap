// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <HelooWorldApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    data-testid="test-title"
  >
    Hola Mundo
  </div>
  <p>
     
    No hay subtitulo
  </p>
  <p>
     
    Trunks
  </p>
</div>
`;
